engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java

```diff
@@ -298,11 +322,23 @@ private void registerDecisionRequirementsAppliers(final MutableProcessingState s
         new DecisionRequirementsDeletedApplier(state.getDecisionState()));
   }
 
+  private void registerDecisionEvaluationAppliers() {
+    register(DecisionEvaluationIntent.EVALUATED, NOOP_EVENT_APPLIER);
+    register(DecisionEvaluationIntent.FAILED, NOOP_EVENT_APPLIER);
+  }
+
   private void registerFormAppliers(final MutableProcessingState state) {
     register(FormIntent.CREATED, new FormCreatedApplier(state.getFormState()));
     register(FormIntent.DELETED, new FormDeletedApplier(state.getFormState()));
   }
 
+  private void registerCompensationSubscriptionAppliers(final MutableProcessingState state) {
+    register(CompensationSubscriptionIntent.CREATED, NOOP_EVENT_APPLIER);
+    register(CompensationSubscriptionIntent.TRIGGERED, NOOP_EVENT_APPLIER);
+    register(CompensationSubscriptionIntent.COMPLETED, NOOP_EVENT_APPLIER);
+    register(CompensationSubscriptionIntent.DELETED, NOOP_EVENT_APPLIER);
```

Comment:
❌ Similar to the change on 8.2, we must fail here rather than apply without state changes. Because on `main` these events produce state changes.

Reply:
Good catch, fixed in c89406b